docs_src/design/adr/0027-URIs for Files.md

```diff
@@ -0,0 +1,113 @@
+    ```    
+
+The type of authentication as well as the credentials will be contained in the secret data specified by the Secret Name. Three types of authentication will be supported, which are `usernamepassword`, `apikey` and `bearertoken`. 
+
+- When `usernamepassword` is specified as the type in the secret data, the **basic-auth** `username:password@` will be inserted into the URI using the **username** and **password** found in the secret data.
+
+    !!! example - "Example secret data - `usernamepassword`"
+        ```
+        type=usernamepassword
+        username=myuser
+        password=mypassword
+        ```
+    !!! example  - "Example resulting URI with `basic-auth`"
+        ```
+        https://myuser:mypassword@example.com/uom.yaml
+        ```
+
+- When `apikey` is specified as the type in the secret data,  the **API Key** will be placed in the HTTP header using `apikeyname` and `apikeyvalue`  from the secret data
+
+    !!! example - "Example secret data - `apikey`"
+        ```
+        type=apikey
+        apikeyvalue=mykeyvalue
+        apikeyname=myname
+        ```
+    
+- When `bearertoken` is specified as the type in the secret data, the **Bearer Token** will be placed in the HTTP header as `Authorization` with value of `Bearer <token>`  where `token` is from the secret data. 
+
+    !!! example - "Example secret data - `token`"
+        ```
+        type=bearertoken
+        token=mytoken
+        ```
+
+### Services/Files Impacted
+
+- **All Services** will be impacted for enabling the loading the **common configuration** and **private configuration** files using URIs. This will be handled in `go-mod-bootstrap's` processing of the `-cc/--commonConfig` and `-cf/--configFile` command line flags.
+
+- **Core Metadata's** loading of the **UOM file** will be adjusted to use the new file load function.
+
+- **Device Service's** loading of **device profiles**, **device definitions** and **provision watchers** files will be adjusted to load an index file specified by a URI in place of the configured folder name. The contents of the index file will be used to load the individual files by URI  by appending the filenames to the original URI. Any authentication specified in the original URI will be used in the subsequent URIs. 
+
+    !!! example - "Example DevicesDir configuration in service configuration" 
+        ```toml
+        [Device]
+          ...
+          ProfilesDir = "./res/profiles"
+          DevicesDir = "http://example.com/devices/index.json"
+          ProvisionWatchersDir = "./res/provisionwatchers"
+          ...
+        ```
+    !!! example - "Example Device Index file `http://example.com/devices/index.json`"
+        ```json
+        [
+            "device1.yaml", "device2.yaml"
+        ]
+        ```
+
+    !!! example - "Example resulting device file URIs from above example"
+        ```
+        http://example.com/devices/device1.yaml
+        http://example.com/devices/device2.yaml

```

Comment:
they should be device1.json and devic2.json for this example

Reply:
Same response as above